flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbTypeInformedElementConverter.java

```diff
@@ -0,0 +1,380 @@
+import org.apache.flink.api.java.typeutils.ObjectArrayTypeInfo;
+import org.apache.flink.api.java.typeutils.PojoTypeInfo;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.api.java.typeutils.TupleTypeInfo;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.connector.dynamodb.table.converter.ArrayAttributeConverterProvider;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.FlinkRuntimeException;
+import org.apache.flink.util.Preconditions;
+
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeValueType;
+import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.internal.mapper.BeanAttributeGetter;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.StaticTableSchema;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * A {@link ElementConverter} that converts an element to a {@link DynamoDbWriteRequest} using
+ * TypeInformation provided.
+ */
+@PublicEvolving
+public class DynamoDbTypeInformedElementConverter<inputT>
+        implements ElementConverter<inputT, DynamoDbWriteRequest> {
+    private final CompositeType<inputT> typeInfo;
+    private final boolean ignoreNulls;
+    private TableSchema<inputT> tableSchema;
+
+    /**
+     * Creates a {@link DynamoDbTypeInformedElementConverter} that converts an element to a {@link
+     * DynamoDbWriteRequest} using the provided {@link CompositeType}. Usage: {@code new
+     * DynamoDbTypeInformedElementConverter<>(TypeInformation.of(MyPojoClass.class))}
+     *
+     * @param typeInfo The {@link CompositeType} that provides the type information for the element.
+     */
+    public DynamoDbTypeInformedElementConverter(CompositeType<inputT> typeInfo) {
+        this(typeInfo, true);
+    }
+
+    public DynamoDbTypeInformedElementConverter(
+            CompositeType<inputT> typeInfo, boolean ignoreNulls) {
+        this.typeInfo = typeInfo;
+        this.ignoreNulls = ignoreNulls;
+    }
+
+    @Override
+    public void open(Sink.InitContext context) {
+        try {
+            tableSchema = createTableSchema(typeInfo);
+        } catch (IntrospectionException | IllegalStateException | IllegalArgumentException e) {
+            throw new FlinkRuntimeException("Failed to extract DynamoDb table schema", e);
+        }
```

Comment:
We should do this in the constructor as a validation and fail fast. Do not need to store, but nice to fail on client and not when running job